From 499ed8b2c081621334fc4816509e8fa7bbb5d677 Mon Sep 17 00:00:00 2001
From: Rohan Puri <rohan_puri@xyratex.com>
Date: Thu, 3 Oct 2013 18:02:03 +0530
Subject: [PATCH 099/159] Fixed 2 bugs: - 1. stobsink related code in
 addb_pfom.c should not be called in kernel 2. stats service merged with
 addb-monitor-infra branch had caused one bug i.e. stats service unregistering
 after successful registeration in stats_svc_init() was happening, due to
 wrong merge, fixed it.

---
 addb/addb_pfom.c  |  3 +++
 stats/stats_srv.c | 10 +---------
 2 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/addb/addb_pfom.c b/addb/addb_pfom.c
index b8d80d4..f574573 100644
--- a/addb/addb_pfom.c
+++ b/addb/addb_pfom.c
@@ -260,9 +260,12 @@ static int addb_pfom_fo_tick(struct m0_fom *fom)
 		if (err != 0)
 			M0_LOG(M0_ERROR, "addb summary posting failed");
 
+/** Only needed for stobsink, so should not be called in kernel */
+#ifndef __KERNEL__
 		if (reqh->rh_addb_mc.am_sink->rs_skulk != NULL)
 			(*reqh->rh_addb_mc.am_sink->rs_skulk)
 				(&reqh->rh_addb_mc);
+#endif
 		m0_fom_phase_set(fom, ADDB_PFOM_PHASE_CTO);
 		break;
 	default:
diff --git a/stats/stats_srv.c b/stats/stats_srv.c
index 9aa3486..00b1f25 100644
--- a/stats/stats_srv.c
+++ b/stats/stats_srv.c
@@ -299,19 +299,11 @@ M0_REQH_SERVICE_TYPE_DEFINE(m0_stats_svc_type, &stats_service_type_ops,
  */
 M0_INTERNAL int m0_stats_svc_init(void)
 {
-	int rc;
-
 	m0_addb_ctx_type_register(&m0_addb_ct_stats_svc);
 	m0_addb_ctx_type_register(&m0_addb_ct_stats_update_fom);
 	m0_addb_ctx_type_register(&m0_addb_ct_stats_query_fom);
 
-	rc = m0_reqh_service_type_register(&m0_stats_svc_type);
-	if (rc != 0)
-		return rc;
-
-	m0_reqh_service_type_unregister(&m0_stats_svc_type);
-
-	return rc;
+	return m0_reqh_service_type_register(&m0_stats_svc_type);
 }
 
 M0_INTERNAL void m0_stats_svc_fini(void)
-- 
1.8.3.2

